source/best-practices/zsh.rst

```diff
@@ -0,0 +1,158 @@
+zsh 及其配置
+============
+
+:本节贡献者: |田冬冬|\（作者）、
+             |姚家园|\（作者）
+:最近更新日期: 2021-01-21
+:预计阅读时间: 20 分钟
+
+Linux 下有很多 Shell，最常见的是 bash，此外还有 zsh、csh、ksh。推荐使用 zsh。
+
+zsh 有如下特点：
+
+- 语法基本完全兼容于 bash，一般用户完全体会不到其区别
+- zsh 提供命令补全特性，比 bash 的补全要更好用
+- 可配置性强
+
+安装 zsh
+---------
+
+在终端键入 ``zsh --version``，若显示 zsh 版本号，则表示 zsh
+已安装。否则需要安装 zsh。
+
+Fedora 用户::
+
+    $ sudo dnf install zsh
+
+CentOS 用户::
+
+    $ sudo yum install zsh
+
+Ubuntu/Debian 用户::
+
+    $ sudo apt install zsh
+
+自 macOS Catalina (10.15) 开始，系统默认的 Shell 已由 bash 变为 zsh。
+
+通过如下命令设置默认 Shell 为 zsh::
+
+    $ chsh -s $(which zsh)
+
+``chsh`` 命令修改的是登陆后的 Shell，因而需要退出当前用户并重新登陆，
+用户的默认 Shell 就从 bash 变成了 zsh 了。打开新的终端键入
+``echo $SHELL``\ ，查看当前 Shell，会显示 :file:`/bin/zsh`\ 。
+
+zsh 的配置文件为 :file:`~/.zshrc`\ 。因而切换到 zsh 后，
+所有的 Shell 配置都不用写到 :file:`~/.bashrc`\ ，而要写到 :file:`~/.zshrc` 中。
+
+Oh My Zsh
+----------
+
+zsh 稍作配置会更加方便好用，`Oh My Zsh <https://ohmyz.sh/>`__ 是一群人
+一起维护的一套 zsh 配置文件，使用起来非常方便。一般用户直接使用该配置即可。
+
+安装 Oh My Zsh::
+
+    $ sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+
+该命令会下载 Oh My Zsh 到 :file:`~/.oh-my-zsh` 目录下，生成默认的 zsh 配置文件 :file:`~/.zshrc`\ ，
+并备份老的配置到类似 :file:`.zshrc.pre-oh-my-zsh` 的文件中。
+
+之后可以根据个人习惯修改配置文件 :file:`.zshrc`\ ，如\
+`设置主题 <https://github.com/ohmyzsh/ohmyzsh#themes>`__\ 
+（如 bira）、\ `启用插件 <https://github.com/ohmyzsh/ohmyzsh#plugins>`__\
+等。
```

Comment:
I feel this plugin list is better, but I don't know if the list is complete or not.

https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins-Overview

Reply:
I put the link https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins at Plugins subsection.
Here, we may want to let the reader know how to use plugins instead of a plugin list.

Reply:
So here, do we want to use the introduction to plugins (https://github.com/ohmyzsh/ohmyzsh#plugins)?

Reply:
Yes.